### ISSUE TITLE
[REQUEST]: Video codec for Oculus Quest 2

### ISSUE THREAD

**Issue by karasusan** [Unity-Technologies/com.unity.webrtc#838]
### Is your feature request related to a problem?

I'm always frustrated when [...]

**Comment by shofixti244** [comment]
Just commenting to say this is an important issue for me.

**Comment by dhruvaura** [comment]
Yes this is a most important issue for many developer. Eagerly waiting for this get fix.

**Comment by chirag-aura** [comment]
Waiting for this solution. This can solve performance issue of Meta Quest when we building some streaming solutions. Request to implement codec asap.

**Comment by half-oval** [comment]
Waiting for this solution. This can solve performance issue of Meta Quest when we building some streaming solutions. Request to implement codec asap.

**Comment by karasusan** [comment]
memo: WRS-378

**Comment by karasusan** [comment]
Same issue #876.

**Comment by kar-re** [comment]
Hi @karasusan!
I'm quite interested in this fix as well, and I am interested in contributing to fix this. 
Do you have any pointers regarding where to look or what to implement in order to mitigate this issue? 

BR,
Kar-re

**Comment by karasusan** [comment]
@kar-re 
The problem is that the codecs implemented by MediaCodec cannot be referenced on Meta Quest 2.
Android codec factory class is implemented by [AndroidCodecFactoryHelper](https://github.com/Unity-Technologies/com.unity.webrtc/blob/main/Plugin~/WebRTCPlugin/Android/AndroidCodecFactoryHelper.cpp).

This class calls the MediaCodec API via a Java class. First, it is necessary to investigate whether the information expected by this API can be obtained.

**Comment by kar-re** [comment]
I see, thank you for the info @karasusan!
I'll give it a go and submit a PR if I manage to get it working.

**Comment by kar-re** [comment]
Hi everyone! Just wanted to share some findings.
Unfortunately I haven't been able to solve the issue in its entirety, but I managed to build a version of the application I'm developing without the stuttering I experienced previously. I thought about writing a custom VideoEncoder for webrtc based on [this repo](https://github.com/Fernthedev/HollywoodQuest/blob/master/shared/render/mediacodec_encoder.hpp), but too much work for initial experimentation. 

The steps taken by me were:
1. Build the native android plugin on my own
2. Change from Vulkan to OpenGLES3 (It crashed with vulkan) 

I've tried to experiment with different codecs and it seems to affect the streamed performance. For my use case it's not really dependent on the recieved fps, but it might be worth knowing. 
[Here is my built .aar if anyone wants to experiment. ](https://drive.google.com/file/d/1KlTVr64l6deJvMG1s1TqIXHXNJ0qmuyD/view?usp=sharing)

This also makes me think it has to do with vulkan compatability somehow.

**Comment by dhruvaura** [comment]
@kar-re 
Curious to know,
Above mentioned improvement will fixed [this unity render stream performance issue](https://github.com/Unity-Technologies/UnityRenderStreaming/issues/716)?

**Comment by dhruvaura** [comment]
Above mentioned improvement will fixed https://github.com/Unity-Technologies/UnityRenderStreaming/issues/716?
